packages/domain-picker/src/domain-picker/use-your-domain-item.tsx

```diff
@@ -15,7 +15,9 @@ interface Props {
 
 const UseYourDomainItem: React.FunctionComponent< Props > = ( { onClick } ) => {
 	return (
-		<label className="domain-picker__suggestion-item contains-link">
+		/* eslint-disable jsx-a11y/click-events-have-key-events */
+		/* eslint-disable jsx-a11y/interactive-supports-focus */
+		<div role="button" className="domain-picker__suggestion-item type-link" onClick={ onClick }>
```

Comment:
This looks very hacky (and it looked hacky as well prior to this PR).

- Why are we not using a `<button />`? Is it because there's another `<button />` nested inside?
- In general, nesting interactive HTML elements within each other (e.g button inside a label, or button inside a button) goes against the HTML spec
- We may want to consider that [the rest of the domain suggestion items have a different wrapping HTML element based on a prop](https://github.com/Automattic/wp-calypso/blob/fae0931f950bd423f99496549638d9ee09e42a0a/packages/domain-picker/src/domain-picker/suggestion-item.tsx#L44-L57) — should we have this component behave the same way, so that it's always aligned with the rest of the items?